src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -88,7 +93,13 @@ public String getWebUrlForWiki(String wikiName) {
 
   public String appendQueryStringToURL(String url, String qs) {
     String separator = url.contains("?") ? "&" : "?";
-    return url + separator + qs;
+
+    String[] filteredUrl = url.split("#");
+    if(filteredUrl.length>1){
+      return filteredUrl[0] + separator + qs + "#" + filteredUrl[1];
```

Comment:
I would added an unit test for this logic here:
https://github.com/Wikia/selenium-tests/blob/master/src/test/java/com/wikia/webdriver/testcases/adstests/TestUrlBuilder.java